src/cdk/v2/destinations/the_trade_desk/utils.js

```diff
@@ -26,82 +33,220 @@ const getSignatureHeader = (request, secretKey) => {
-  return [response, ...errorResponseList];
+const prepareCustomProperties = (message, destination) => {
+  const { customProperties } = destination.Config;
+  const payload = {};
+  if (customProperties) {
+    customProperties.forEach((customProperty) => {
+      const { rudderProperty, tradeDeskProperty } = customProperty;
+      const value = get(message, rudderProperty);
+      if (value) {
+        payload[tradeDeskProperty] = value;
+        lodash.unset(message, rudderProperty);
+      }
+    });
+  }
+  return payload;
 };
 
-const processRouterDest = (inputs) => {
-  const respList = [];
-  const { destination } = inputs[0];
-  const groupedInputs = lodash.groupBy(inputs, (input) => input.message.type);
-  if (groupedInputs.record) {
-    const transformedRecordEvent = processRecordInputs(groupedInputs.record, destination);
-    respList.push(...transformedRecordEvent);
+const populateEventName = (message, destination) => {
+  let eventName;
+  const { event } = message;
+  const { eventsMapping } = destination.Config;
+
+  if (Array.isArray(eventsMapping) && eventsMapping.length > 0) {
+    const keyMap = getHashFromArray(eventsMapping, 'from', 'to');
+    eventName = keyMap[event.toLowerCase()];
+  }
+
+  if (eventName) {
+    return eventName;
+  }
+
+  const eventMapInfo = ECOMM_EVENT_MAP[event.toLowerCase()];
+  if (isDefinedAndNotNull(eventMapInfo)) {
+    return eventMapInfo.event;
   }
 
-  return respList;
+  return event;
 };
 
-module.exports = { getSignatureHeader, processRouterDest };
+/**
+ * Retrieves the data processing options based on the provided message.
+ *
+ * @param {string} message - The message to process.
+ * @throws {InstrumentationError} - Throws an error if the region is not supported, if no policies are provided, if multiple policies are provided, or if the policy is not supported.
+ * @returns {Object} - The data processing options, including the policies and region.
+ */
+const getDataProcessingOptions = (message) => {
+  const integrationObj = getIntegrationsObj(message, 'THE_TRADE_DESK') || {};
+  let { policies } = integrationObj;
+  const { region } = integrationObj;
+  let dataProcessingOptions;
+
+  if (region && !region.toLowerCase().startsWith('us')) {
+    throw new InstrumentationError('Only US states are supported');
+  }
+
+  if (!policies || (Array.isArray(policies) && policies.length === 0)) {
+    policies = ['LDU'];
+  }
+
+  if (policies.length > 1) {
+    throw new InstrumentationError('Only one policy is allowed');
+  }
+
+  if (policies[0] !== 'LDU') {
+    throw new InstrumentationError('Only LDU policy is supported');
+  }
+
+  if (policies && region) {
+    dataProcessingOptions = { policies, region };
+  }
+
+  return dataProcessingOptions;
+};
+
+const getPrivacySetting = (message) => {
+  const integrationObj = getIntegrationsObj(message, 'THE_TRADE_DESK');
+  return integrationObj?.privacy_settings;
```

Comment:
Do we need to have any value checks here?

Reply:
Not needed right now. will add later upon any feedback